Basic_viewer/doc/Basic_viewer/CGAL/Qt/Basic_viewer_qt.h

```diff
@@ -0,0 +1,256 @@
+namespace CGAL {
+
+//------------------------------------------------------------------------------
+/*!
+ \ingroup PkgBasicViewerClasses
+
+The class `Basic_viewer_qt` is a Qt widget based on `QGLViewer` that allows to visualize 3D elements: points, segments, triangles, rays and lines. This class stores a reference to a `Graphics_scene`. Elements can either be added directly in the viewer or through the scene. This class requires `CGAL_Qt5`, and is only available if the macro `CGAL_USE_BASIC_VIEWER` is defined. Linking with the cmake target `CGAL::CGAL_Basic_viewer` will link with `CGAL_Qt5` and add the definition `CGAL_USE_BASIC_VIEWER`.
+
+\tparam BufferType the number type used for point coordinates: `float` by default.
+
+*/
+template <typename BufferType=float>
+class Basic_viewer_qt : public CGAL::QGLViewer
```

Comment:
`BufferType` is the wrong name. A buffer is an array. A "buffer type" could be `std::vector<double>`, but not double. You are looking for the name of the type of elements in the buffer. I suggest `DataType`.

Reply:
I wrote for the same reason:  "Why is it not NumberType. Also can it be anything else than double or float? Maybe Precision ?"

My guess is that it can only be `float` or `double` ,   so it could also be `FloatingPointType".

Reply:
@afabri and @lrineau I suggest to remove the template parameter from the doc.
Indeed 99.99% of times, we use flot.
And I can keep it in the code, undocumented, just in case we need later some specific uses.
Do you agree?

Reply:
The class `Graphics_scene` also has it. so why would you remove it for `Basic_viewer_qt`?
You did not reply to my question if it can be something else than float or double.
Also please build the documentation again.

Reply:
I suggest to remove it from all classes (graphics scene and basic viewer qt).

/forcebuild:v20-10-2023

Reply:
Stop documenting things that are not the reality.

If `Graphics_scene` and `Basic_viewer_qt` are class templates, even with a default template argument, you cannot document them as if they were just (non-template) classes.

I agree those classes should not be templates at all. The implementation cannot work, for the moment, for anything but `float` as the data type for the buffer. The reason is that is a call to [`QOpenGLShaderProgram::setAttributeBuffer`](https://doc.qt.io/qt-6/qopenglshaderprogram.html#setAttributeBuffer) one has to choose what is the `type` argument (`GL_FLOAT` or `GL_DOUBLE`). See for example line 404 here:
https://github.com/CGAL/cgal/blob/1e364417e669eac2e0bb2670a172894af49a6b8e/Basic_viewer/include/CGAL/Qt/Basic_viewer_qt.h#L400-L408

Actually, the switch between `float` or `double` to represent the coordinates could be delayed to runtime. The graphics scene would have to use tricks to have the correct `Buffer_for_vao<Data_type>>` for members, but that could be doable. That could be useful for scenes that have a bounding box entirely far from the origin. But for the moment, I think you should stick to `float`, without any template parameter. That is what the implementation expects, anyway.

Reply:
done